seldon-core-operator/tox.ini

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+[tox]
+skipsdist = True
+skip_missing_interpreters = True
+
+[testenv]
+setenv =
+    PYTHONPATH={toxinidir}
+    PYTHONBREAKPOINT=ipdb.set_trace
+    CHARM_REPO=https://github.com/canonical/seldon-core-operator.git
+    CHARM_BRANCH=main
+    LOCAL_CHARM_DIR=charm_repo
+
+[testenv:prepare-rock]
+passenv = *
+allowlist_externals =
+    bash
+    git
+    skopeo
+    tox
+    rm
+    rockcraft
+    yq
+setenv =
+    # Needed for juju cli to work correctly
+    HOME={env:HOME}
+deps =
+    juju~=2.9.0
+    pytest
+    pytest-operator
+    ops
+commands =
+    # build and pack rock
+    rockcraft pack
+    # upload rock to docker and microk8s cache, replace charm's container with local rock reference
+    bash -c 'NAME=$(yq eval .name rockcraft.yaml) && \
```

Comment:
Why are you using `bash -c '...'` here and not just type the commands one by one like the rest of the `tox.ini` files?

Tox will run the commands one by one and will fail if one fails, so shouldn't it be the same?
https://tox.wiki/en/latest/config.html#commands

Reply:
`tox` does not do evaluation. Those commands will not set variables as desired.

Reply:
yeah @kimwnasptd if you know how to do something like
```bash
X=1
echo $X
```
in tox, it would make things much cleaner!  But neither of us have figured that out :)